### PR TITLE
Fix stuff with diagonals

### LINKED ISSUE #1042: `ÞD` does not work correctly on non-square matrices
[Try it Online!](https://vyxal.pythonanywhere.com/#WyJQIiwiIiwi4p+o4p+oIDEgfCAyIHwgMyB8IDQg4p+pIHxcbiDin6ggNSB8IDYgfCA3IHwgOCDin6kgfFxuIOKfqCA5IHwgMTAgfCAxMSB8IDEyIOKfqeKfqSAgw55EXG5cbiMgc2hvdWxkIHJldHVybjogW1sxLDYsMTFdLCBbMiw3LDEyXSwgWzMsOF0sIFs0XSwgWzldLCBbNSwxMF1dIiwiIiwiIl0=) and [Try it Online!](https://vyxal.pythonanywhere.com/#WyJQIiwiIiwi4p+o4p+oIDEgIHwgMiAgfCAzIOKfqSB8XG4g4p+oIDQgIHwgNSAgfCA2IOKfqSB8XG4g4p+oIDcgIHwgOCAgfCA5IOKfqSB8XG4g4p+oIDEwIHwgMTEgfCAxMiDin6nin6kgIMOeRFxuXG4jIHNob3VsZCByZXR1cm46IFtbMSw1LDldLCBbMiw2XSwgWzNdLCBbMTBdLCBbNywxMV0sIFs0LDgsMTJdXSIsIiIsIiJd)

### LINKED ISSUE #1043: Add element to get all anti-diagonals of matrix

### PR DESCRIPTION
**Pull request by Steffan153**
Fixes #1042 
Fixes #1043

### CODE REVIEW

**Review by Seggan — APPROVED** [review]

File: vyxal/elements.py [hunk 1]
```diff
@@ -215,15 +215,32 @@ def all_diagonals(lhs, ctx):
     Diagonals of a matrix, starting with the main diagonal.
     """
     vector = [iterable(x, ctx=ctx) for x in lhs]
-    all_diags = [[] for _ in range(len(vector) * 2 - 1)]
+    all_diags = [[] for _ in range(len(vector) + len(vector[0]) - 1)]
```
Comment (cgccuser) [inline vyxal/elements.py:218]:
```suggestion
    if not vector:
        return []
    all_diags = [[] for _ in range(len(vector) + len(vector[0]) - 1)]
```

File: vyxal/elements.py [hunk 2]
```diff
@@ -215,15 +215,32 @@ def all_diagonals(lhs, ctx):
...
             all_diags[(start + i) % len(all_diags)].append(row[i])
         start -= 1
     return all_diags
 
 
+def all_antidiagonals(lhs, ctx):
+    """Element Þḋ
+    Anti-diagonals of a matrix, starting with the main anti-diagonal.
+    """
+    vector = [iterable(x, ctx=ctx) for x in lhs]
```
Comment (cgccuser) [inline vyxal/elements.py:231]:
```suggestion
    vector = [iterable(x, ctx=ctx) for x in lhs]
    if not vector:
        return []
```

File: vyxal/elements.py [hunk 3]
```diff
@@ -1145,7 +1163,7 @@ def diagonal(lhs, ctx):
     (any) -> diagonal of a
     """
     lhs = [iterable(elem, ctx=ctx) for elem in iterable(lhs, ctx=ctx)]
-    return [lhs[i][i] for i in range(len(lhs))]
+    return [lhs[i][i] for i in range(min(len(lhs), len(lhs[0])))]
```
Comment (cgccuser) [inline vyxal/elements.py:1166]:
```suggestion
    if not lhs:
        return []
    return [lhs[i][i] for i in range(min(len(lhs), len(lhs[0])))]
```

**Review by cgccuser — APPROVED** [review]
LGTM, just some suggestions to check for empty lists

**Review by cgccuser — CHANGES_REQUESTED** [review]
Grr

**Comment by cgccuser** [comment]
@Steffan153 You can add the suggestions to a batch and commit them all at once btw. You don't need to do each separately

**Review by cgccuser — APPROVED** [review]

**Pull request merged**